core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -937,4 +938,17 @@ default HttpResponse recover(Function<? super Throwable, ? extends HttpResponse>
         requireNonNull(function, "function");
         return of(new RecoverableStreamMessage<>(this, function, /* allowResuming */ false));
     }
+
+    default HttpResponse recover(Class<? extends Throwable> causeClass,
+                                 Function<? super Throwable, ? extends HttpResponse> function) {
+        requireNonNull(causeClass, "causeClass");
+        requireNonNull(function, "function");
+        final StreamMessage<HttpObject> stream = recover(cause -> {
+            if (cause.getClass().equals(causeClass)) {
+                return function.apply(cause);
```

Comment:
We raise an exception if the function returns `null`: https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/internal/common/stream/RecoverableStreamMessage.java#L222
How about also applying the same approach here?

Reply:
Oh. good idea
I understood that if the function.apply(e) returns null, it should be handled accordingly.
Is that correct?

Reply:
That's correct. 😉